x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/ReplaceAggregateNestedExpressionWithEval.java

```diff
@@ -159,10 +125,73 @@ protected LogicalPlan rule(Aggregate aggregate) {
             aggregate = aggregate.with(newEval, groupings, aggregates);
         }
 
-        return (LogicalPlan) aggregate;
+        return aggregate;
+    }
+
+    private static Expression transformNonEvaluatableGroupingFunction(
+        GroupingFunction.NonEvaluatableGroupingFunction gf,
+        List<Alias> evals
+    ) {
+        int[] counter = new int[] { 0 };
+        boolean childrenChanged = false;
+        List<Expression> newChildren = new ArrayList<>(gf.children().size());
+
+        for (Expression ex : gf.children()) {
+            if (ex instanceof Attribute == false) { // TODO: foldables shouldn't require eval'ing either
```

Comment:
What is this TODO about? Are `NonEvaluatableGroupingFunction` expected to be potentially foldable?

Reply:
I think we shouldn't need to have to transform a `... BY CATEGORIZE("value")` into `EVAL ref = "value" | ... BY CATEGORIZE(ref)`